### PR TITLE
 dotCMS/core#16445 Error when change a page template

### PR DESCRIPTION
**Pull request by freddyucv**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java [hunk 1]
```diff
@@ -631,6 +631,20 @@ private void addEmptyContainers(final IHTMLPage page,
         }
     }
 
+    private boolean isContains(
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java]: Please add some Javadoc here, the method's name is not clear, or change it

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java [hunk 2]
```diff
@@ -631,6 +631,20 @@ private void addEmptyContainers(final IHTMLPage page,
...
+            final Table<String, String, Set<String>> pageContents,
+            final ContainerUUID containerUUID, Container container) {
+
+        final boolean isLegacyValue = ContainerUUID.UUID_LEGACY_VALUE.equals(containerUUID.getUUID());
+        boolean pageContenstContains = pageContents.contains(containerUUID.getIdentifier(), ContainerUUID.UUID_START_VALUE);
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java]: Typo here.

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java (hunk 1 above)
Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java]: rename to contains

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java [hunk 3]
```diff
@@ -618,9 +619,8 @@ private void addEmptyContainers(final IHTMLPage page,
 
 
                 if (!pageContents.contains(container.getIdentifier(), containerUUID.getUUID())) {
-                    final boolean isLegacyValue = ContainerUUID.UUID_LEGACY_VALUE.equals(containerUUID.getUUID());
 
-                    if (!isLegacyValue || !pageContents.contains(containerUUID.getIdentifier(), ContainerUUID.UUID_START_VALUE)) {
+                    if (doesPageContentsHaveContainer(pageContents, containerUUID, container)) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/factories/MultiTreeAPIImpl.java]: Doesn't this change require updating / adding test(s)?

**Review by jdotcms — APPROVED** [review]

**Review by dsilvam — APPROVED** [review]

**Pull request merged**